dataactcore/scripts/pullFPDSData.py

```diff
@@ -0,0 +1,1158 @@
+        # params = 'SIGNED_DATE:[2017/03/01,'+ yesterday.strftime('%Y/%m/%d') + '] '
+    # if a date that the script was last successfully run is provided, get data since that date
+    else:
+        last_run_date = last_run.update_date
+        params = 'LAST_MOD_DATE:[' + last_run_date.strftime('%Y/%m/%d') + ',' + yesterday.strftime('%Y/%m/%d') + '] '
+
+    # TODO remove this later, this is just for testing
+    # params += 'CONTRACTING_AGENCY_ID:1542 '
+    # params = 'VENDOR_ADDRESS_COUNTRY_CODE:"GBR"'
+    # params = 'PIID:"0046"+REF_IDV_PIID:"W56KGZ15A6000"'
+
+    i = 0
+    logger.info('Starting: ' + feed_url + params + 'CONTRACT_TYPE:"' + contract_type.upper() +
+                '" AWARD_TYPE:"' + award_type + '"')
+    while True:
+        resp = requests.get(feed_url + params + 'CONTRACT_TYPE:"' + contract_type.upper() + '" AWARD_TYPE:"' +
+                            award_type + '"&start=' + str(i), timeout=60)
+        resp_data = xmltodict.parse(resp.text, process_namespaces=True,
+                                    namespaces={'http://www.fpdsng.com/FPDS': None,
+                                                'http://www.w3.org/2005/Atom': None})
+        # only list the data if there's data to list
+        try:
+            listed_data = list_data(resp_data['feed']['entry'])
+        except KeyError:
+            listed_data = []
+
+        for ld in listed_data:
+            data.append(ld)
+            i += 1
+
+        # Log which one we're on so we can keep track of how far we are (different numbers for different pulls)
+        if i % log_interval == 0:
+            logger.info("On line " + str(i))
+
+            # every 100,000 records, add the data to the session and clear the data array so we don't run into
+            # a memory error. This can be done inside the 100 tracker because 100,000 is divisible by 100
+            # so we don't have to check this if statement every time. Don't process on row 0
+            if i % 100000 == 0 and i != 0:
+                logger.info("Processing next 100,000 records")
+                process_and_add(data, contract_type, sess, last_run)
+                data = []
+
+        # if we got less than 10 records, we can stop calling the feed
+        if len(listed_data) < 10:
+            break
+
+    logger.info(i)
+
+    # insert whatever is left
+    process_and_add(data, contract_type, sess, last_run)
+    logger.info("processed " + contract_type + ": " + award_type + " data")
+
+
+def get_delete_data(contract_type, now, sess, last_run):
+    """ Get data from the delete feed """
+    data = []
+    yesterday = now - datetime.timedelta(days=1)
+    last_run_date = last_run.update_date
+    params = 'LAST_MOD_DATE:[' + last_run_date.strftime('%Y/%m/%d') + ',' + yesterday.strftime('%Y/%m/%d') + '] '
+
+    i = 0
+    logger.info('Starting: ' + delete_url + params + 'CONTRACT_TYPE:"' + contract_type.upper() + '"')
```

Comment:
See comment above on line 994 about adding specificity as to which (get vs delete) the start message refers to.